tests/ignite/metrics/test_recall.py

```diff
@@ -493,184 +231,63 @@ def to_numpy_multilabel(y):
-    assert re1.compute() == pytest.approx(re2.compute().mean().item())
-
-
-def test_multilabel_input_NCHW():
-    def _test(average):
-        re = Recall(average=average, is_multilabel=True)
-
-        y_pred = torch.randint(0, 2, size=(10, 5, 18, 16))
-        y = torch.randint(0, 2, size=(10, 5, 18, 16)).long()
-        re.update((y_pred, y))
-        np_y_pred = to_numpy_multilabel(y_pred)
-        np_y = to_numpy_multilabel(y)
         assert re._type == "multilabel"
         re_compute = re.compute() if average else re.compute().mean().item()
         with warnings.catch_warnings():
             warnings.simplefilter("ignore", category=UndefinedMetricWarning)
             assert recall_score(np_y, np_y_pred, average="samples") == pytest.approx(re_compute)
 
-        re.reset()
-        y_pred = torch.randint(0, 2, size=(10, 4, 20, 23))
-        y = torch.randint(0, 2, size=(10, 4, 20, 23)).long()
-        re.update((y_pred, y))
-        np_y_pred = to_numpy_multilabel(y_pred)
-        np_y = to_numpy_multilabel(y)
-        assert re._type == "multilabel"
-        re_compute = re.compute() if average else re.compute().mean().item()
-        with warnings.catch_warnings():
-            warnings.simplefilter("ignore", category=UndefinedMetricWarning)
-            assert recall_score(np_y, np_y_pred, average="samples") == pytest.approx(re_compute)
-
-        # Batched Updates
-        re.reset()
-        y_pred = torch.randint(0, 2, size=(100, 5, 12, 14))
-        y = torch.randint(0, 2, size=(100, 5, 12, 14)).long()
-
-        batch_size = 16
-        n_iters = y.shape[0] // batch_size + 1
-
-        for i in range(n_iters):
-            idx = i * batch_size
-            re.update((y_pred[idx : idx + batch_size], y[idx : idx + batch_size]))
-
-        np_y = to_numpy_multilabel(y)
-        np_y_pred = to_numpy_multilabel(y_pred)
-        assert re._type == "multilabel"
-        re_compute = re.compute() if average else re.compute().mean().item()
-        with warnings.catch_warnings():
-            warnings.simplefilter("ignore", category=UndefinedMetricWarning)
-            assert recall_score(np_y, np_y_pred, average="samples") == pytest.approx(re_compute)
+        re1 = Recall(is_multilabel=True, average=True)
+        re2 = Recall(is_multilabel=True, average=False)
+        re1.update((y_pred, y))
+        re2.update((y_pred, y))
+        assert re1.compute() == pytest.approx(re2.compute().mean().item())
+
+    def get_test_cases():
+
+        test_cases = [
+            (torch.randint(0, 2, size=(10, 5)), torch.randint(0, 2, size=(10, 5)), 1),
+            (torch.randint(0, 2, size=(10, 4)), torch.randint(0, 2, size=(10, 4)), 1),
+            # updated batches
+            (torch.randint(0, 2, size=(50, 5)), torch.randint(0, 2, size=(50, 5)), 16),
+            (torch.randint(0, 2, size=(50, 4)), torch.randint(0, 2, size=(50, 4)), 16),
+            (torch.randint(0, 2, size=(10, 5, 10)), torch.randint(0, 2, size=(10, 5, 10)), 1),
```

Comment:
Please, add a comment here that it is another test case and not `# updated batches` anymore.

Reply:
Thanks! that's done now!